pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -395,6 +396,23 @@ func (w *watcher) execute(ctx context.Context, repo git.Repo, repoID string, eve
 					})
 					continue
 				}
+				if handler.Config.MakePullRequest {
+					retry := backoff.NewRetry(retryPushNum, backoff.NewConstant(retryPushInterval))
```

Comment:
There is a redundant here, since we also push later (same file L450) after this switch statement. Previously, when only have single branch to push, the follow is

- first, loop through all eventCfgs (start at L339), handle each event one by one and commit the changes to local branch
- end the loop, do push to the remote branch (since there is only one branch to watch/commit) to reduce the network cost by multi push commit

As we changed the behavior, and we agree that: users can set whether an event change should commit to the watching branch or create a new branch at event level (since an application configuration may contain a list of eventWatcher handler). We will face problem of having multiple branches locally that need to be push at the end of the eventCfgs for loop.

So I think we have to think about this carefully:
- How should we push the commited branch to the remote
- How should we handle update event handled status

Reply:
Thanks so much!
I fixed in this PR( [fix how to push new branch](https://github.com/pipe-cd/pipecd/pull/4395/commits/20222a3e996139b3e7a96ed5e8f6523710ca486c) ).
- Newly created branches are now pushed together after loop.
- The default branch is pushed at the end as before.

I could not come up with the best solution for the following.
Please let me know if you have a better solution🙇‍♂️

> - How should we push the committed branches to the remote
> - How should we handle updating event handled status

Reply:
Thank for contribution, I committed to the latest code 🙏